CollapseLauncher/XAMLs/MainApp/Pages/SettingsPage.xaml

```diff
@@ -348,6 +348,49 @@
                         <TextBlock Margin="0,16,0,8"
                                    Style="{ThemeResource SubtitleTextBlockStyle}"
                                    Text="{x:Bind helper:Locale.Lang._SettingsPage.NetworkSettings_Title}" />
+                        <StackPanel Margin="0,0,0,16">
+                            <TextBlock FontSize="16"
+                                       Style="{ThemeResource SubtitleTextBlockStyle}"
+                                       Text="Custom DNS Settings" />
+                            <ToggleSwitch x:Name="CustomDnsSettingsToggle"
+                                          IsOn="{x:Bind DnsSettingsContext.IsUseExternalDns, Mode=TwoWay}"
+                                          OffContent="{x:Bind helper:Locale.Lang._Misc.Disabled}"
+                                          OnContent="{x:Bind helper:Locale.Lang._Misc.Enabled}" />
+                            <Grid Width="400"
+                                  HorizontalAlignment="Left"
+                                  RowSpacing="8">
+                                <Grid.RowDefinitions>
+                                    <RowDefinition Height="Auto"/>
+                                    <RowDefinition Height="Auto"/>
+                                    <RowDefinition Height="Auto"/>
+                                </Grid.RowDefinitions>
+                                <ComboBox ItemsSource="{x:Bind DnsSettingsContext.ExternalDnsConnectionTypeList}"
+                                          SelectedIndex="{x:Bind DnsSettingsContext.ExternalDnsConnectionType, Mode=TwoWay}"
+                                          IsEnabled="{x:Bind CustomDnsSettingsToggle.IsOn, Mode=OneWay}"
+                                          HorizontalAlignment="Stretch"
+                                          Header="Connection Type"/>
+                                <ComboBox Grid.Row="1"
+                                          ItemsSource="{x:Bind DnsSettingsContext.ExternalDnsProviderList}"
+                                          SelectedIndex="{x:Bind DnsSettingsContext.ExternalDnsProvider, Mode=TwoWay}"
+                                          IsEnabled="{x:Bind CustomDnsSettingsToggle.IsOn, Mode=OneWay}"
+                                          HorizontalAlignment="Stretch"
+                                          Header="DNS Provider"/>
+                                <TextBox x:Name="CustomDnsHostTextbox"
+                                         Grid.Row="2"
+                                         Text="{x:Bind DnsSettingsContext.ExternalDnsAddresses, Mode=TwoWay}"
+                                         IsEnabled="{x:Bind CustomDnsSettingsToggle.IsOn, Mode=OneWay}"
+                                         Header="Custom DNS Provider Host(s)"
+                                         Visibility="Collapsed"/>
```

Comment:
Needs a "Test/Validate & Save" button and also a tooltip/help button to help user write the setting